presto-main/src/test/java/com/facebook/presto/block/BlockAssertions.java

```diff
@@ -33,19 +40,176 @@
 import static com.facebook.presto.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
 import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
 import static com.facebook.presto.testing.TestingConnectorSession.SESSION;
+import static com.facebook.presto.type.TypeJsonUtils.appendToBlockBuilder;
+import static com.facebook.presto.type.TypeUtils.checkElementNotNull;
+import static java.lang.Double.isNaN;
 import static java.util.Objects.requireNonNull;
 import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
 
 public final class BlockAssertions
 {
-    private BlockAssertions()
+    private BlockAssertions() {}
```

Comment:
Contains a bunch of code copied from all the structural type.equalTos
Is more permissive in general that equalTo, as tests rely on null values and NaNs comparing intuitively.